File: SDKs/Node/Common/src/types/providerType.ts
/**
 * Strongly-typed enumeration of supported LLM providers.
 * These numeric values must match the C# ProviderType enum exactly.
 * @see https://github.com/knnlabs/Conduit/blob/main/ConduitLLM.Core/Enums/ProviderType.cs
 */
export enum ProviderType {
  /** OpenAI provider (GPT models) */
  OpenAI = 1,
  
  /** Anthropic provider (Claude models) */
  Anthropic = 2,
  
  /** Azure OpenAI Service */
  AzureOpenAI = 3,
  
  /** Google Gemini */
  Gemini = 4,
  
  /** Google Vertex AI */
  VertexAI = 5,
  
  /** Cohere */
  Cohere = 6,
  
  /** Mistral AI */
  Mistral = 7,
  
  /** Groq */
  Groq = 8,
  
  /** Ollama (local models) */
  Ollama = 9,
  
  /** Replicate */
  Replicate = 10,
  
  /** Fireworks AI */
  Fireworks = 11,
  
  /** AWS Bedrock */
  Bedrock = 12,
  
  /** Hugging Face */
  HuggingFace = 13,
  
  /** AWS SageMaker */
  SageMaker = 14,
  
  /** OpenRouter */
  OpenRouter = 15,
  
  /** OpenAI-compatible generic provider */
  OpenAICompatible = 16,
  
  /** MiniMax */
  MiniMax = 17,
  
  /** Ultravox */
  Ultravox = 18,
  
  /** ElevenLabs (audio) */
  ElevenLabs = 19,
  
  /** Google Cloud (audio) */
  GoogleCloud = 20,
  
  /** Cerebras (high-performance inference) */
  Cerebras = 21
}

/**
 * Type guard to check if a value is a valid ProviderType
 */
export function isProviderType(value: unknown): value is ProviderType {
  return typeof value === 'number' && 
         value >= ProviderType.OpenAI && 
         value <= ProviderType.Cerebras;
}

/**
 * Get the display name for a provider type
 */
export function getProviderDisplayName(provider: ProviderType): string {
  const names: Record<ProviderType, string> = {
    [ProviderType.OpenAI]: 'OpenAI',
    [ProviderType.Anthropic]: 'Anthropic',
    [ProviderType.AzureOpenAI]: 'Azure OpenAI',
    [ProviderType.Gemini]: 'Google Gemini',
    [ProviderType.VertexAI]: 'Google Vertex AI',
    [ProviderType.Cohere]: 'Cohere',
    [ProviderType.Mistral]: 'Mistral AI',
    [ProviderType.Groq]: 'Groq',
    [ProviderType.Ollama]: 'Ollama',
    [ProviderType.Replicate]: 'Replicate',
    [ProviderType.Fireworks]: 'Fireworks AI',
    [ProviderType.Bedrock]: 'AWS Bedrock',
    [ProviderType.HuggingFace]: 'Hugging Face',
    [ProviderType.SageMaker]: 'AWS SageMaker',
    [ProviderType.OpenRouter]: 'OpenRouter',
    [ProviderType.OpenAICompatible]: 'OpenAI Compatible',
    [ProviderType.MiniMax]: 'MiniMax',
    [ProviderType.Ultravox]: 'Ultravox',
    [ProviderType.ElevenLabs]: 'ElevenLabs',
    [ProviderType.GoogleCloud]: 'Google Cloud',
    [ProviderType.Cerebras]: 'Cerebras'
  };
  
  return names[provider] || 'Unknown Provider';
}

File: Shared/ConduitLLM.Persistence.Abstractions/Entities/ProviderType.cs
namespace ConduitLLM.Configuration;

/// <summary>
/// Strongly-typed enumeration of supported LLM providers. This identifies an
/// adapter type, not a unique configured provider instance.
/// </summary>
public enum ProviderType
{
    Unknown = 0,
    OpenAI = 1,
    Groq = 2,
    Replicate = 3,
    Fireworks = 4,
    OpenAICompatible = 5,
    MiniMax = 6,
    Ultravox = 7,
    ElevenLabs = 8,
    Cerebras = 9,
    SambaNova = 10,
    DeepInfra = 11,
    Cloudflare = 12,
    OpenRouter = 13,
    Meta = 14,
    Azure = 15,
    Bedrock = 16,
    Vertex = 17
}

/// <summary>
/// Defines the provider types that can be used for provider configuration.
/// </summary>
public static class ProviderTypeCatalog
{
    /// <summary>
    /// All provider types backed by an operational provider adapter.
    /// </summary>
    public static IReadOnlyList<ProviderType> ConfigurableTypes { get; } = Array.AsReadOnly(
        Enum.GetValues<ProviderType>()
            .Where(providerType => providerType != ProviderType.Unknown)
            .ToArray());

    /// <summary>
    /// Returns whether the value identifies a configurable provider adapter.
    /// </summary>
    public static bool IsConfigurable(ProviderType providerType) =>
        providerType != ProviderType.Unknown && Enum.IsDefined(providerType);
}
